### PR TITLE
Added service worker with PWA capability

### PR DESCRIPTION
**Pull request by ishythefishy**

### CODE REVIEW

File: apps/fretonator-web/src/index.html [hunk 1]
```diff
@@ -1,42 +1,46 @@
 <!DOCTYPE html>
 <html lang="en">
-  <head>
+<head>
+  <script async="" src="https://www.googletagmanager.com/gtag/js?id=UA-167450036-1"></script>
+  <script>
```
Comment (geometricpanda) [inline apps/fretonator-web/src/index.html:5]: Isn’t it best to put GTM in the body? So it doesn’t block the render?

File: apps/fretonator-web/src/index.html [hunk 2]
```diff
@@ -1,42 +1,46 @@
...
 
-    <meta charset="utf-8" />
-    <title>Fretonator</title>
-    <base href="/" />
-    <meta name="viewport" content="width=device-width, initial-scale=1" />
+  <meta charset="utf-8"/>
+  <title>Fretonator</title>
+  <base href="/"/>
+  <meta name="viewport" content="width=device-width, initial-scale=1"/>
+  <meta property="og:type" content="website">
```
Comment (geometricpanda) [inline apps/fretonator-web/src/index.html:21]: There’s a Meta service you can inject into your components to do this bit, then on each ngOnInit you can change it relevant to the route you’re on

File: apps/fretonator-web/src/index.html [hunk 3]
```diff
@@ -1,42 +1,46 @@
...
-  <body>
-    <app-root></app-root>
-  </body>
+  <link rel="icon" type="image/x-icon" href="favicon.ico"/>
+  <link rel="manifest" href="manifest.webmanifest">
+  <meta name="theme-color" content="#1a1a1a">
+</head>
+<body>
+<app-root></app-root>
+<noscript>I know it sucks, but please enable JavaScript to continue using this application.</noscript>
```
Comment (geometricpanda) [inline apps/fretonator-web/src/index.html:44]: As you’re prerendering your app, if we change the knowledge of which route you’re on to be url based, you can get this to work without JS

File: apps/fretonator-web/src/index.html (hunk 1 above)
Comment (ishythefishy) [reply apps/fretonator-web/src/index.html:5]: Ugh true. They always just say put it as high in the head as possible.

File: apps/fretonator-web/src/index.html (hunk 2 above)
Comment (ishythefishy) [reply apps/fretonator-web/src/index.html:21]: ah didn't know it scaled to all the things

**Pull request merged**